File: tournament-winner/TournamentWinner.java
import java.util.*;

public class TournamentWinner {
   //1 - Home | 0 - Away
   private static final Integer HOME_TEAM_WON = 1;
   private static final Integer AWAY_TEAM_WON = 0;
   private static final Integer COUNT = 3;
   Map<String, Integer> scores = new HashMap<>();
   String winner = new String();
   public String tournamentWinner(ArrayList<ArrayList<String>> competitions, ArrayList<Integer> results) {
     for(int i = 0; i < competitions.size(); i++){
       ArrayList<String> competition = competitions.get(i);
       String winningTeam = competition.get(results.get(i).equals(AWAY_TEAM_WON) ? HOME_TEAM_WON : AWAY_TEAM_WON);
       Integer currentScore = scores.getOrDefault(winningTeam, 0) + COUNT;
       scores.put(winningTeam, currentScore);
       winner = scores.getOrDefault(winner, 0) < currentScore ? winningTeam : winner;
     }
     return winner;
   }
}

File: tournament-winner/TournamentWinner.js
var tournamentWinner = (competitions, results) => {
    const AWAY_TEAM_WON = 0;
    const HOME_TEAM_WON = 1;
    const COUNT = 3;
    let scores = new Map();
    let winner = "";
    for(let index = 0; index < competitions.length; index++) {
        let competition = competitions[index];
        let winningTeam = competition[results[index] === AWAY_TEAM_WON ? HOME_TEAM_WON : AWAY_TEAM_WON];
        scores.set(winningTeam, (scores.get(winningTeam) ?? 0) + COUNT);
        winner = (scores.get(winner) ?? 0) < scores.get(winningTeam) ? winningTeam : winner;
    }
    return winner;
}

exports.tournamentWinner = tournamentWinner;